ports/llfio/portfile.cmake

```
@@ -0,0 +1,78 @@
+message(WARNING [=[
+LLFIO depends on Outcome which depends on QuickCppLib which uses the vcpkg versions of gsl-lite and byte-lite, rather than the versions tested by QuickCppLib's, Outcome's and LLFIO's CI. It is not guaranteed to work with other versions, with failures experienced in the past up-to-and-including runtime crashes. See the warning message from QuickCppLib for how you can pin the versions of those dependencies in your manifest file to those with which QuickCppLib was tested. Do not report issues to upstream without first pinning the versions as QuickCppLib was tested against.
+]=])
+
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ned14/llfio
+    REF 9294f05d48220a6a5d6d307da5fe8f236dd020cf
+    SHA512 00bd49fe568ba2a840917de8d5b8d398d7480b2431f99f8c3b282fc60785d8664869fd83e8ac56ffc9a1eaa0b991c5d8a97ab5e53e20e5567f3d75577ad4feb4
+    HEAD_REF develop
+)
+
+if(WIN32)
+  vcpkg_from_github(
+      OUT_SOURCE_PATH NTKEC_SOURCE_PATH
+      REPO ned14/ntkernel-error-category
+      REF bbd44623594142155d49bd3ce8820d3cf9da1e1e
+      SHA512 589d3bc7bca98ca8d05ce9f5cf009dd98b8884bdf3739582f2f6cbf5a324ce95007ea041450ed935baa4a401b4a0242c181fb6d2dcf7ad91587d75f05491f50e
+      HEAD_REF master
+  )
+endif()
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS LLFIO_FEATURE_OPTIONS
+    FEATURES
+      status-code LLFIO_USE_EXPERIMENTAL_SG14_STATUS_CODE
+      run-tests LLFIO_ENABLE_DEPENDENCY_SMOKE_TEST
+)
+
+# LLFIO needs a copy of QuickCppLib with which to bootstrap its cmake
+file(COPY "${CURRENT_INSTALLED_DIR}/include/quickcpplib"
+    DESTINATION "${SOURCE_PATH}/quickcpplib/repo/include/"
+)
+file(COPY "${CURRENT_INSTALLED_DIR}/share/ned14-internal-quickcpplib/"
+    DESTINATION "${SOURCE_PATH}/quickcpplib/repo/"
+)
+
+# LLFIO expects ntkernel-error-category to live inside its include directory
+file(REMOVE_RECURSE "${SOURCE_PATH}/include/llfio/ntkernel-error-category")
+if(WIN32)
+  file(RENAME "${NTKEC_SOURCE_PATH}" "${SOURCE_PATH}/include/llfio/ntkernel-error-category")
+endif()
+
+vcpkg_configure_cmake(
+    SOURCE_PATH "${SOURCE_PATH}"
+    PREFER_NINJA
+    OPTIONS
+        -DPROJECT_IS_DEPENDENCY=On
+        -Dquickcpplib_FOUND=1
+        -Doutcome_FOUND=1
+        "-DCMAKE_CXX_FLAGS=-I${CURRENT_INSTALLED_DIR}/include"
+        ${LLFIO_FEATURE_OPTIONS}
+        -DLLFIO_ENABLE_DEPENDENCY_SMOKE_TEST=ON  # Leave this always on to test everything compiles
+        -DCMAKE_DISABLE_FIND_PACKAGE_Git=ON
+)
+
+# Build the shared library and static library editions
+vcpkg_build_cmake(TARGET _sl)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    vcpkg_build_cmake(TARGET _dl)
+endif()
```

```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
    vcpkg_build_cmake(TARGET _dl)
else()
    vcpkg_build_cmake(TARGET _sl)
endif()
```
I'm not entirely sure if this is the correct fix, however vcpkg ports must produce only one variant (static/shared) of libraries. The variant requested by the user is defined by `VCPKG_LIBRARY_LINKAGE`.

`vcpkg_configure_cmake()` will also automatically define the standard CMake variable for requesting only one flavor: `BUILD_SHARED_LIBS` (https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html)

---

As you've probably noticed, none of my libraries observe `BUILD_SHARED_LIBS`, they always generate targets for header only, static, and shared libraries (and indeed entire target trees for each of the sanitisers and static analysers for each of those again). LLFIO very strongly encourages the static library build over all others, and everything assumes it is always built, because it always is. All target trees generate uniquely named files, nothing ever collides, so apart from it taking slightly longer it is benign.